spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/result/AbstractResult.java

```diff
@@ -68,4 +70,56 @@ public boolean failed() {
     public String getFailureDetail() {
         return failure == null ? null : String.join(", ", getFailureMessages());
     }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} is successful
+     */
+    public void ifSuccess(Consumer<T> successAction) {
+        if (succeeded()) {
+            successAction.accept(getContent());
+        }
+    }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} failed. Passes the {@link Failure} to the consumer
+     */
+    public void ifFailure(Consumer<F> failureAction) {
+        if (failed()) {
+            failureAction.accept(getFailure());
+        }
+    }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} is successful. Returns this {@link Result} to enable fluent statements.
+     *
+     * @return this
+     */
+    public AbstractResult<T, F> whenSuccessful(Consumer<AbstractResult<T, F>> content) {
+        if (succeeded()) {
+            content.accept(this);
+        }
+        return this;
+    }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} failed. Passes {@code this} result to the consumer.
+     */
+    public void orElseDo(Consumer<AbstractResult<T, F>> result) {
+        if (failed()) {
+            result.accept(this);
+        }
+    }
+
+    /**
+     * Throws an exception supplied by the {@link Supplier} if this {@link Result} is not successful.
+     *
+     * @param exceptionSupplier provides an instance of the exception to throw
+     */
+    public <X extends Throwable> void orElseThrow(Supplier<? extends X> exceptionSupplier) throws X {
```

Comment:
It could return `this` as well

Reply:
sounds legit, would only be useful in the non-throw case though.